crates/forge-runner/src/test_case_summary.rs

```diff
@@ -121,6 +152,58 @@ impl TestCaseSummary<Fuzzing> {
             _ => None,
         }
     }
+
+    #[must_use]
+    pub fn from(results: Vec<TestCaseSummary<Single>>) -> TestCaseSummary<Fuzzing> {
+        let last: TestCaseSummary<Single> = results
+            .iter()
+            .last()
+            .cloned()
+            .expect("Fuzz test should always run at least once");
+        // Only the last result matters as fuzzing is cancelled after first fail
+        match last {
+            TestCaseSummary::Passed {
+                name,
+                msg,
+                arguments,
+                gas_info: _,
+                test_statistics: (),
+            } => {
+                let runs = results.len();
+                let gas_usages_vec: Vec<u128> = results
+                    .into_iter()
+                    .filter(|item| matches!(item, TestCaseSummary::Passed { .. }))
```

Comment:
Do we need this line? From my understanding, they should all be `Passed` if `last` is passed.